### DISCUSSION TITLE
Good practice for multiple setters

### DISCUSSION THREAD

**Discussion by edoudou** [facebookexperimental/Recoil#2013] [Q&A] [answered]
What are the good practice to implement multiple setter for 1 selector ?

I would like to create multiple setters for a single getter, should I : 
- Create multiple selectors with the same setter ?
- Use one selector with an `{ action, data }` in put and use a switch statement to handle my many setters. 

In my case I am working on a edition tool with drag and drop elements, I have atoms representing parents containing a list of children ids, and each children is created as an atom from a family.
I have a selector that send me back my parent atom with all its children. I would like to create different setter such as Add, Remove, UpdateMany, ...

**Comment by drarmstr** [comment, marked as answer]
For setters you can also use `useRecoilCallback()` or `useRecoilTransaction_UNSTABLE()` for different actions.

**Reply by edoudou** [reply]
Yes I ended up using `recoil-nexus` which is using `useRecoilCallback` internally.
